tests/Tests/Fixtures/BaseFixtureManager.php

```diff
@@ -63,4 +85,132 @@ public function getUnregisteredUuid()
+                    array_push($sqlBinds, $fieldValue);
+                }
+            }
+            $sqlColumnValues = rtrim($sqlColumnValues, " ,");
+            $isInserted = QueryUtils::sqlInsert($sqlInsert . $sqlColumnValues, $sqlBinds);
+            if ($isInserted) {
+                $insertCount += 1;
+            }
+        }
+        return $insertCount;
+    }
+
+    public function installFixtures()
+    {
+        $fixtures = $this->getFixturesFromFile();
+        $insertCount = $this->installFixturesForTable($this->tableName, $fixtures);
+        $this->hasInstalledFixtured = true;
+        return $insertCount;
+    }
+
+    // we don't have a good way to solve this so we will force all sub classes to implement.
+    public function removeFixtures()
+    {
+        // we have no generic way of signifying what the primary key of the table is so we force sub classes
+        // to implement the remove fixture method
+        $this->removeInstalledFixtures();
+        $this->hasInstalledFixtured = false;
+    }
+
+    public function hasInstalledFixtures()
+    {
+        return $this->hasInstalledFixtured;
+    }
+
+    /**
+     * @return a random fixture.
+     */
+    public function getSingleFixture()
+    {
+        $fixtures = $this->getFixturesFromFile();
+        return $this->getSingleEntry($fixtures);
+    }
+
+    abstract protected function removeInstalledFixtures();
+
+    private function getValueFromFunction($value)
+    {
+        $functionName = strtok($value, "()");
+
+        // white list our functions here
+        if ($functionName === "uuid") {
+            $table = strtok("(,)");
+            if ($table !== false) {
+                // trim spaces, and quotes
+                $table = trim($table, "'\" \t\n\r\0\x0B");
+                return $this->getUuid($table);
+            } else {
+                throw new \BadMethodCallException("uuid(table_name) function is missing table name");
+            }
+        } else if ($functionName === "generateId") {
+            return QueryUtils::generateId();
+        } else {
+            throw new \BadMethodCallException("Function could not be interpreted from fixture: " . $value);
+        }
+        return ""; // return empty string
+    }
+
+    private function isFunctionCall($value)
+    {
+        return preg_match("/[a-zA-Z]+\(['a-zA-Z_0-9\-]*\)/", $value) === 1;
+    }
+
+    private function isForeignReference(array $reference)
+    {
+        return isset($reference['table']) && isset($reference['columnSearch']) && isset($reference['columnSearchValue']) && isset($reference['columnReference']);
+    }
+
+    private function getQueryForForeignReference(array $reference): SearchQueryFragment
+    {
+        // these values come from flat files in the unit tests, I'm torn on whether we should escape these or not
+        // TODO: @adunsulag check with @brady.miller on whether we should escape these or not
```

Comment:
@bradymiller These are only used in the unit tests, I've been wondering if it really matters to escape these tables since the json files are part of the codebase.

Reply:
rec escaping them, since never know where this stuff will lead. Plus will be nice to fail if for some reason the the requested table/column does not exist anymore for some reason.

Reply:
again, though, no need to go way overboard on escaping here if it overcomplicates things :)

Reply:
This should be fixed now.